content/en/docs/zero-code/java/agent/extensions.md

```diff
@@ -49,8 +68,1083 @@ extensions:
+                      error.getClass().getSimpleName());
+      }
+    }
+  }
+}
+```
+
+### ConfigurablePropagatorProvider
+
+Register custom propagators that can be referenced by name in the
+`otel.propagators` configuration.
+
+**Example:**
+
+```java
+import com.google.auto.service.AutoService;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigurablePropagatorProvider;
+
+@AutoService(ConfigurablePropagatorProvider.class)
+public class MyPropagatorProvider implements ConfigurablePropagatorProvider {
+
+  @Override
+  public TextMapPropagator getPropagator(ConfigProperties config) {
+    return new MyCustomPropagator();
+  }
+
+  @Override
+  public String getName() {
+    // Use with: -Dotel.propagators=my-propagator
+    return "my-propagator";
+  }
+}
+```
+
+### ConfigurableSamplerProvider
+
+Register custom samplers that can be referenced in the `otel.traces.sampler`
+configuration.
+
+**Example:**
+
+```java
+import com.google.auto.service.AutoService;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.autoconfigure.spi.traces.ConfigurableSamplerProvider;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+
+@AutoService(ConfigurableSamplerProvider.class)
+public class MySamplerProvider implements ConfigurableSamplerProvider {
+
+  @Override
+  public Sampler createSampler(ConfigProperties config) {
+    // Access configuration
+    double samplingRate = config.getDouble("otel.traces.sampler.arg", 1.0);
+    return new MyCustomSampler(samplingRate);
+  }
+
+  @Override
+  public String getName() {
+    // Use with: -Dotel.traces.sampler=my-sampler
+    return "my-sampler";
+  }
+}
+```
+
+### ResourceProvider
+
+Add custom resource attributes that will be automatically merged with other
+resource providers.
+
+**Example:**
+
+```java
+import com.google.auto.service.AutoService;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.autoconfigure.spi.ResourceProvider;
+import io.opentelemetry.sdk.resources.Resource;
+
+@AutoService(ResourceProvider.class)
+public class MyResourceProvider implements ResourceProvider {
+
+  @Override
+  public Resource createResource(ConfigProperties config) {
+    return Resource.create(
+        Attributes.builder()
+            .put("deployment.environment", "production")
+            .put("service.version", getVersion())
+            .put("custom.attribute", "value")
+            .build()
+    );
+  }
+
+  private String getVersion() {
+    // Retrieve version from environment, file, etc.
+    return System.getenv().getOrDefault("APP_VERSION", "unknown");
+  }
+}
+```
+
+## Application-Agent Communication
+
+### Accessing the Current Span
+
+In instrumentation advice classes, use `Java8BytecodeBridge` to access the
```

Comment:
Should we explain why and when one needs to use `Java8BytecodeBridge` instead of `Span.current()` or `Context.current()`?
Jumping to advice while skipping `InstrumentationModule` and `TypeInstrumentation` feels a bit confusing.